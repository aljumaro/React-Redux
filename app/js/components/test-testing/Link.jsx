import React from 'react';

const STATUS = {
	hovered: 'hovered',
	normal: 'normal'
};

export default class Link extends React.Component {

	constructor(props) {
		super(props);

		this._onMouseEnter = this._onMouseEnter.bind(this);
		this._onMouseLeave = this._onMouseLeave.bind(this);

		this.state = {
			className: STATUS.normal
		}
	}

	_onMouseEnter() {
		this.setState({className: STATUS.hovered});
	}

	_onMouseLeave() {
		this.setState({className: STATUS.normal});
	}

	render() {
		return (
			<a
				className={this.state.className}
				href={this.props.page || '#'}
				onMouseEnter={this._onMouseEnter}
				onMouseLeave={this._onMouseLeave}
			>
				{this.props.children}
			</a>
		)
	}

}
