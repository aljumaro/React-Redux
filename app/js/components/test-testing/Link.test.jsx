import React from 'react';
import Link from './Link';
import renderer from 'react-test-renderer';

it('renders link correctly', () => {

	const component = renderer.create(<Link page="www.google.es">To google</Link>);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();

	tree.props.onMouseEnter();
	tree = component.toJSON();
  	expect(tree).toMatchSnapshot();

  	tree.props.onMouseLeave();
  	tree = component.toJSON();
  	expect(tree).toMatchSnapshot();

});

it('renders hash correctly', () => {

	const component = renderer.create(<Link>To google</Link>);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});