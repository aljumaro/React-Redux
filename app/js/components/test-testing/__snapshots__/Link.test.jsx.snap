// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders hash correctly 1`] = `
<a
  className="normal"
  href="#"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  To google
</a>
`;

exports[`renders link correctly 1`] = `
<a
  className="normal"
  href="www.google.es"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  To google
</a>
`;

exports[`renders link correctly 2`] = `
<a
  className="hovered"
  href="www.google.es"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  To google
</a>
`;

exports[`renders link correctly 3`] = `
<a
  className="normal"
  href="www.google.es"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  To google
</a>
`;
